Breadcrumbs.py

```diff
@@ -32,80 +38,169 @@ def get_row_indentation(points, view, tab_size, limit=1e20):
-      last_line_start = line_start
-      current_indentation = get_row_indentation(points, view, tab_size, indentation)
-      if current_indentation < indentation and not is_white_row(view, points):
-        indentation = current_indentation
-        breadcrumbs.append(get_breadcrumb(view, points, breadcrumb_length_limit))
+    if current_row < 0:
+      return
 
-      current_row -= 1
+    indentation = get_row_indentation(get_points_by_row(current_row), view, tab_size) + 1
+  else:
+    indentation = get_row_indentation(get_points_by_row(current_row), view, tab_size)
+    current_row -= 1
+
+  breadcrumbs = []
+  last_line_start = view.text_point(current_row + 1, 0)
+  while 0 <= current_row and 0 < indentation:
+    line_start = get_row_start(current_row)
+    points = xrange(line_start, last_line_start)
+    last_line_start = line_start
+    current_indentation = get_row_indentation(points, view, tab_size, indentation)
+    if current_indentation < indentation and not is_white_row(view, points):
+      indentation = current_indentation
+      breadcrumbs.append(get_breadcrumb(view, points, my_regex, separator))
+
+    current_row -= 1
+
+  breadcrumbs.reverse()
+  lengths = [len(breadcrumb) for breadcrumb in breadcrumbs]
+  sorted_lengths = sorted(lengths)
+  previous_length = 0
+  number_of_characters_left  = total_breadcrumbs_length_limit - len(lengths) * len(separator)
+  for (number_of_shorter, current_length) in enumerate(sorted_lengths):
+    if previous_length < current_length:
+      previous_length = current_length
+      number_of_not_shorter = len(breadcrumbs) - number_of_shorter
+      if number_of_characters_left < current_length * number_of_not_shorter:
+        length_of_short_trim = number_of_characters_left // number_of_not_shorter
+        length_of_long_trim = length_of_short_trim + 1
+        number_of_long_trimmed = number_of_characters_left % number_of_not_shorter
+        number_of_short_trimmed = number_of_not_shorter - number_of_long_trimmed
+        for (index_of_breadcrumb, breadcrum_length) in enumerate(lengths):
+          if current_length <= breadcrum_length:
+            if 0 < number_of_short_trimmed:
+              length_of_trim = length_of_short_trim
+              number_of_short_trimmed -= 1
+            else:
+              length_of_trim = length_of_long_trim
+            breadcrumbs[index_of_breadcrumb] = breadcrumbs[index_of_breadcrumb][0:length_of_trim]
+        break
+    number_of_characters_left -= current_length
+
+  return ''.join(breadcrumbs+[''])
+
+
+stylesheet = '''
+  <style>
+    div.phantom-arrow {
+      border-top: 0.4rem solid transparent;
+      border-left: 0.5rem solid color(var(--bluish) blend(var(--background) 30%));
+      width: 0;
+      height: 0;
+    }
+    div.phantom {
+      padding: 0.4rem 0 0.4rem 0.7rem;
+      margin: 0 0 0.2rem;
+      border-radius: 0 0.2rem 0.2rem 0.2rem;
+      background-color: color(var(--bluish) blend(var(--background) 30%));
```

Comment:
This thread is informative: https://forum.sublimetext.com/t/dev-build-3118/21270
The idea is that very little (color) styling should be necessary here and everything to make it work is either generated on the fly by Sublime or described in the color scheme. Also, packages can't override each others styling, but a color scheme can make general changes by e.g. changing the value of `--greenish` or by targeting a specific phantom using the id's. All of that seems pretty impractical though, I don't expect to see color scheme authors going through the effort to tweak phantoms.